tasks/connectors/netsparker/lib/netsparker_client.rb

```diff
@@ -44,14 +47,16 @@ def retrieve_all_scheduled_ids
         def get_last_scan_id(schedule_id)
           found = nil
           page = 1
+          response = http_get(list_scheduled_url(page), @headers)
+          scheduled_scan_result = JSON.parse(response)
+
```

Comment:
That's better, but we still fetch all scheduled ids for every entry in @scheduled_ids